docs/data/toolpad/core/components/persistent-state/use-session-storage-state-api.md

```diff
@@ -0,0 +1,41 @@
+# useSessionStorageState API
+
+<p class="description">API reference for the useSessionStorageState hook.</p>
+
+:::success
+For examples and details on the usage of this React hook, visit the demo pages:
+
+- [useSessionStorageState](/toolpad/core/react-persistent-state/)
+
+:::
+
+## Import
+
+```js
+import useSessionStorageState from '@toolpad/core/useSessionStorageState';
+// or
+import { useSessionStorageState } from '@toolpad/core';
+```
+
+Learn about the difference by reading this [guide](https://mui.com/material-ui/guides/minimizing-bundle-size/) on minimizing bundle size.
+
+## Reference
+
+### `useSessionStorageState`
+
+```js
+const [state, setState] = useSessionStorageState('my-key', 'initial value');
+```
+
+**Parameters**
+
+- `key`: `string | null` The key under which to store the value in `window.sessionStorage`.
```

Comment:
Doing API pages manually lets us explain things a bit better in some cases, maybe I might do the same for some components such as the `AppProvider`.

Reply:
Even the hooks generation for base is really just for base specific hooks, they won't work for ours